### PR TITLE
[5.1] Allow different namespace roots within RouteGroups

### PR DESCRIPTION
**Pull request by hpolthof**
This will fix the behaviour as described below.

Imagine a package with a service provider that creates a Router macro to include all the required routes for the package.
Like this:

```
$this->app['router']->macro('package', function () {
    $this->group(['prefix' => 'package', 'namespace' => '\Vendor\Package\Http\Controllers'], function ($router) {
        $router->get('demo', 'DemosController@getDemo');
    });
});
```

Then in the `routes.php` file this macro will be called in the way described below:

```
Route::group(['prefix' => 'admin', 'namespace' => 'Admin'], function() {
    Route::get('dashboard', 'HomeController@getDashboard');

    Route::package();
});
```

This should result in the following routes:

```
/admin/dashboard    -> App\Http\Controllers\Admin\HomeController@getDashboard
/admin/package/demo -> Vendor\Package\Http\Controllers\DemosController@getDemo 
```

but in fact this results in:

```
/admin/dashboard    -> App\Http\Controllers\Admin\HomeController@getDashboard
/admin/package/demo -> App\Http\Controllers\Admin\Vendor\Package\Http\Controllers\DemosController@getDemo 
```

As a result of this the controllers outside of the current scope are not found.

Therefor it should be possible to break out of the namespace stack when the namespace is explicitly prefix with a '\' symbol.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Broken build.

**Comment by GrahamCampbell** [comment]
Also, please add a test.

**Comment by hpolthof** [comment]
This should do the trick, to pass all the tests.

**Comment by GrahamCampbell** [comment]
No merge commits please.

File: src/Illuminate/Routing/Router.php [hunk 1]
```diff
@@ -417,6 +417,13 @@ public static function mergeGroup($new, $old)
     protected static function formatUsesPrefix($new, $old)
     {
         if (isset($new['namespace'])) {
+            // If a namespace is provided prefixed by a '\' character
```
Comment (GrahamCampbell) [inline src/Illuminate/Routing/Router.php]: comments need to be cascading by length - it's a laravel ocd thing, no joke :P

**Pull request closed without merging**